aii-core/src/main/perl/Shellfe.pm

```diff
@@ -323,6 +339,38 @@ sub _initialize
     if ($self->option(INCLUDE)) {
         unshift(@INC, split(/:+/, $self->option(INCLUDE)));
     }
+
+    # Initialization and validation of some options whose default values are based on other options
+    if ( $self->option(NBPDIR_PXELINUX) ) {
+        $self->{CONFIG}->set(NBPDIR_PXELINUX, undef) if $self->option(NBPDIR_PXELINUX) eq NBPDIR_VARIANT_DISABLED;
+    }
+    if ( $self->option(NBPDIR_GRUB2) ) {
+        $self->{CONFIG}->set(NBPDIR_GRUB2, undef) if $self->option(NBPDIR_GRUB2) eq NBPDIR_VARIANT_DISABLED;
+    } else {
+        if ( $self->option(NBPDIR_PXELINUX) ) {
+            my $efi_dir = dirname($self->option(NBPDIR_PXELINUX)) . OSINSTALL_DEF_GRUB2_DIR;
+            $self->{CONFIG}->set(NBPDIR_GRUB2, $efi_dir);
+        } else {
+            $self->{CONFIG}->set(NBPDIR_GRUB2, OSINSTALL_DEF_ROOT_PATH . OSINSTALL_DEF_GRUB2_DIR);
+        }
+    }
+    if ( !$self->option(GRUB2_EFI_KERNEL_ROOT) ) {
+        if ( $self->option(NBPDIR_GRUB2) ) {
+            my $kernel_root = dirname($self->option(NBPDIR_GRUB2));
```

Comment:
so this is what is meant above with `with first and last path tokens removed`? it is not clear, maybe an example like
`/a/b/c/e -> b/c` would clarify things